externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/FetchIterator.scala

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.engine.flink
+
+/**
+ * Borrowed from Apache Spark, see SPARK-33655
+ */
+sealed private[engine] trait FetchIterator[A] extends Iterator[A] {
```

Comment:
We have already extracted the `FetchIterator` into the `kyuubi-common` for multiple engines purpose. Can we reuse it?

Reply:
Removed